plonky2/src/plonk/circuit_builder.rs

```diff
@@ -392,6 +397,48 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         })
     }
 
+    /// Find an available slot, of the form `(gate_index, op)` for gate `G`.
+    pub fn find_slot<G: Gate<F, D> + Clone>(
+        &mut self,
+        gate: G,
+        params: &[F],
```

Comment:
Oh, a comment to explain what `params` is would be helpful here too. Also, very much a nit, but maybe keep `params` and `constants` in the same order here as elsewhere?